### DISCUSSION TITLE
Passing arrays from C# to C++

### DISCUSSION THREAD

**Discussion by jeunathe** [nxrighthere/UnrealCLR#121] [General]
Hi everyone.

I want to be able to send data from c# which is of byte[] type (packed data) and being able to get in to unreal C++ and then unpack it using a reader.
I tried using TArray<uint8> but it seems i fail to transfert this data. I am trying to send it as a string, with no success.

Any idea ?
I am trying to sync a known format packet with a unreal defined Ustruct.

Best regards,

Jnat

**Comment by nxrighthere** [comment]
Arrays are not supported in [data passing](https://github.com/nxrighthere/UnrealCLR/blob/master/MANUAL.md#data-passing) at the moment. This feature requires intermediate conversion logic to reconstruct arrays on the native side from managed code. Most likely this feature [will be available](https://github.com/nxrighthere/UnrealCLR/discussions/112) exclusively in workflow with blueprints.

**Reply by jeunathe** [reply]
ok ! Thank you for the answer !
